Client/Frontend/ContentBlocker/ContentBlockerHelper+Whitelist.swift

```diff
@@ -6,6 +6,35 @@ import WebKit
 import Shared
 import Deferred
 
+class WhitelistedDomains {
+    private(set) var domainSet = Set<String>()
+    private(set) var domainRegex = [NSRegularExpression]()
```

Comment:
could all the methods below clear/add/delete be written in a `didSet` closure for `domainSet` instead?

Reply:
Can you explain how?

Reply:
It  doesnt seem like any of the load/clear/add methods seem to be guarding against any sort of bad data. So we could just allow setting/getting of domainSet. Then other classes would simply do` ContentBlocker.WhitelistedDomains.domainSet.add` which would trigger `didSet` where we could then update the regex.

Reply:
If you it clearer to create an immutable accessor I'll add it to the patch:
```
class WhitelistedDomains {
    private var _domainSet = Set<String>() {
        didSet {
            domainRegex = domainSet.flatMap { wildcardContentBlockerDomainToRegex(domain: "*" + $0) }
        }
    }

    var domainSet: Set<String> {
        return Set<String>(_domainSet)
    }

    private(set) var domainRegex = [NSRegularExpression]()

    func load(_ list: [String]) {
        _domainSet = Set(list)
    }

    func clear() {
        _domainSet = Set<String>()
    }

    func add(_ domain: String) {
        var set = domainSet
        set.insert(domain)
        _domainSet = set
    }

    func delete(_ domain: String) {
        var set = domainSet
        set.remove(domain)
        _domainSet = set
    }
}
```